Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -16,6 +17,178 @@
 
 namespace GradientSignal
 {
+    // Custom AZ::Job so that we can bake the output image asynchronously.
+    // We create the AZ::Job with isAutoDelete = false so that we can detect
+    // when the job has completed, which means we need to handle its deletion.
+    BakeImageJob::BakeImageJob(
+        const GradientBakerConfig& configuration,
+        const AZ::IO::Path& fullPath,
+        AZ::Aabb inputBounds,
+        AZ::EntityId boundsEntityId
+        )
+        : AZ::Job(false, nullptr)
+        , m_configuration(configuration)
+        , m_outputImageAbsolutePath(fullPath)
+        , m_inputBounds(inputBounds)
+        , m_boundsEntityId(boundsEntityId)
+    {
+    }
+
+    void BakeImageJob::Process()
+    {
+        // Get the actual resolution of our image.  Note that this might be non-square, depending on how the window is sized.
+        const int imageResolutionX = aznumeric_cast<int>(m_configuration.m_outputResolution.GetX());
+        const int imageResolutionY = aznumeric_cast<int>(m_configuration.m_outputResolution.GetY());
+
+        // The TGA and EXR formats aren't recognized with only single channel data,
+        // so need to use RGBA format for them
+        int channels = 1;
+        if (m_outputImageAbsolutePath.Extension() == ".tga" || m_outputImageAbsolutePath.Extension() == ".exr")
```

Comment:
Should these checks ignore case? Is something outside of the job enforcing lowercase file names?

Reply:
Good callout! I wasn't sure so I tested and verified that the file dialog that is used to feed the image path forces the extension to lowercase even if you chose a file with upper-case extension (e.g. `.PNG`)